### PR TITLE
Local adjustments - Normalize luminance - improvment Preview TM - Retinex - Remove D.Range

### PR DESCRIPTION
**Pull request by Desmis**
"Normalize luminance" is a process of "IPOL" origin, it allows to obtain an output image which has the same "mean" and the same "standard deviation" as the original, it is notably used after Laplacian and Poisson equation. I implemented it - at least for 2 years - in Tone Mapping, Retinex, and Dynamic Range

In any case it works very well for pipeline processes that process the whole image ("improccordinator", "simpleprocess"), but the operation on the partial image used in "dcrop" [preview zoom] was problematic (bad behavior) on some images.

This PR improve behavior (preview) for "Tone Mapping" and "Retinex", and for "D.Range" this function is removed, in the hope that one day the pipeline will be reviewed ?

Jacques

### CODE REVIEW

File: rtengine/iplocallab.cc [hunk 1]
```diff
@@ -1831,23 +1831,20 @@ float find_gray(float source_gray, float target_gray)
 }
 
 void ImProcFunctions::mean_sig (LabImage* savenorm, float &meanf, float &stdf, int xxs, int xxe, int yys, int yye){
-    int nbf = 0;
+    int size = (yye - yys) * (xxe - xxs);
```
Comment (heckflosse) [inline rtengine/iplocallab.cc]: @Desmis Jacques, can I push a small change to this function?

**Comment by Desmis** [comment]
@heckflosse 

No problem, you can :)

Jacques

**Comment by heckflosse** [comment]
@Desmis Jacques, before I push my changes, I need to know why `mean_sig` is called in improccoordinator.cc, but not in simpleprocess.cc

This sounds like a bug...

**Comment by Desmis** [comment]
@heckflosse 

Ingo, no it is not a bug...it is intentional, because it is to solve the problem of the preview (and only the preview)

I calculate "mean and sigma" in improccoordinator.cc to used these values in dcrop.cc (only)

"normalize_mean_dt" is called "normally" for "improccordinator" and "simpleprocess"...It's only with dcrop wher I used "mean and sigma" calculated in improccoordinator.cc

Jacques

**Comment by Desmis** [comment]
@heckflosse 

Ingo

Thank you :)

Jacques

**Comment by heckflosse** [comment]
De rien.

**Comment by heckflosse** [comment]
@Desmis Jacques, can I push some changes to make iplocallab.cc cppcheck clean again?

**Comment by Desmis** [comment]
No problem :)

Le sam. 27 mars 2021 14:30, Ingo Weyrich ***@***.***> a
écrit :

**Comment by Desmis** [comment]
@heckflosse 

Ingo
Do you have other improvements to make ? if not, I propose to merge this PR, and of course if yes..I wait

Jacques

**Comment by heckflosse** [comment]
@Desmis Jacques, I will make one more change today

**Comment by Desmis** [comment]
@heckflosse 

Ingo
OK :)

jacques

**Comment by heckflosse** [comment]
@Desmis Done!

**Comment by Desmis** [comment]
@heckflosse 

Ingo

Can I merge ? Or I wait ?

Jacques

**Comment by heckflosse** [comment]
You can merge

**Comment by Desmis** [comment]
@heckflosse 

Ingo

Thank you :)

Jacques

**Pull request merged**